gwcs/tests/test_wcs.py

```diff
@@ -1056,3 +1056,19 @@ def test_tabular_2d_quantity():
 
     bb = w.bounding_box
     assert all(u.allclose(u.Quantity(b), [0, 2] * u.pix) for b in bb)
+
+
+@pytest.mark.filterwarnings("error:.*Indexing a WCS.pipeline step is deprecated.*:DeprecationWarning")
+def test_initialize_wcs_with_list():
+    # test that you can  initialize a wcs with a pipeline that is a list
+    # containing both Step() and (frame, transform) tuples
+
+    trans = models.Shift(10 * u .pix) & models.Shift(2 * u.pix)
+    pipeline = [('detector', trans), ('sky', None)]
+
+    # add a Step() to the pipeline list
+    extra_step = wcs.Step('extra_step', trans)
+    pipeline.append(extra_step)
```

Comment:
This test creates the pipeline incorrectly. The `None` sentinel must be in the last step. The errors is currently raised only when the WCS is printed or executed. There should be a check in the initialization of the WCS.